src/persist-client/src/batch.rs

```diff
@@ -684,6 +690,7 @@ where
         if num_updates == 0 {
             return;
         }
+        let diffs_sum = diffs_sum::<D>(std::iter::once(&columnar)).expect("part is non empty");
```

Comment:
I originally had typed this as being computed as updates were added, but I didn't end up seeing any particular benefit to that over this. Switched to computing it in bulk at the end since we'll be using this field in a way that is load-bearing for correctness, and I felt this way it was more obvious that we didn't miss anything in the sum. Flexible, though!